### DISCUSSION TITLE
Meshcentral stoped working overnight and is missing  folder 'node-windows' in  C:\Program Files\Open Source\MeshCentral\node_modules\

### DISCUSSION THREAD

**Discussion by chdodier** [Ylianst/MeshCentral#5695] [Issues] [answered]
[meshcentral.err.log](https://github.com/Ylianst/MeshCentral/files/13933165/meshcentral.err.log)
Restoring or fixing installation issue
meshcentral 2  is using node.js 20.10.0 on Windows Server 2019
My setup was working fine for at least 4 months.
Last night, the windows service stopped working and I found in meshcentral.err log this message : 
errror: Cannot find module 'C:\Program Files\Open Source\MeshCentral\node_modules\node-windows\lib\wrapper.js'
The node-windows folder is not there. I tried reinstalling using MeshCentralInstaller-2.11.exe but the folder was not recreated.
 I do have a backup of the database but it does not include this folder. What would be the proper restore procedure?

**Comment by si458** [comment, marked as answer]
yes i am aware of this issue, i through i fixed the package install
but i think i made it worse for ANY windows users that uses the meshcentral installer, https://github.com/Ylianst/MeshCentral/issues/5694

simple fix at the moment,
1. backup the whole meshcentral folder, and importantly the meshcentral-data folder !?
2. STOP MESHCENTRAL
3. UNINSTALL MESHCENTRAL COMPLETELY! (including delete folders)
4. UNINSTALL NODEJS (no idea why????)
5. REINSTALL MESHCENTRAL (this will install node 14)
6. STOP MESHCENTRAL AGAIN
7. RESTORE meshcentral-data
8. START MESHCENTRAL
9. check all is ok
CARRY ON IF YOU WANT THE LATEST NODEJS
10. STOP MESHCENTRAL
11. INSTALL NODE 20 (tick box asking for extra packages with chocolatey)
12. START MESHCENTRAL

EDIT: im in the process of looking into it with my VM why the installer doesnt upgrade correctly?
also how did you upgrade? was it via web ui? was it automatic? did you run the installer and click update?

**Reply by chdodier** [reply]
Thank You so much it worked. I really did not want to restart from scratch.
To answer you question. I upgraded nodejs from version 14.17.1 to 20.10 using the installer from node's web page. 
I followed the upgrade procedure from nodejs.
But this was done at least a month before meshcentral crashed.
I did it after receiving the warning in my mescentral server tab that my server needed to use v16 at least in the future.
After fixing the server using your procedure I get the message.
And I am a little reluctant in upgrading nodejs for now. But if it can help you solve an issue I will be glad to help.
Last time I upgraded from 14.17.1 to 20.1. Should I Get another version like 16? 

Link to a snapshot of the warning in french...
![Mesh](https://github.com/Ylianst/MeshCentral/assets/141847659/e96adfa7-fbd8-42d6-808f-af84b867e52e)

Also I installed it on Windows 2019 because this is what I am must comfortable with but I dont mind moving this to another platform.
In your experience, witch once would be the most reliable? an is it possible to transfert my database to any other platform ?

**Reply by si458** [reply]
glad its sorted! and was simple!
in theory you can stop meshcentral and upgrade nodejs (use the x64 node 20 msi installer) then restart meshcentral and the warning will go away!

in my personal option, its much easier to run it on linux because you WONT need the extra package 'node-windows' as thats the package that some how magically deleted itself when people who used the meshcentral installer upgraded?

also if you are using the default nedb database, the is built in a migration tool which migrates the nedb to a different database, but sadly the migration tool is only from nedb to say mariadb or mongodb, it wont work from mongodb to mariadb for example

**Comment by isd94techdir** [comment]
I had mine set to auto-update and the same thing happened to me. I'm not opposed to switching to Linux to host my MeshCentral server, but I'm not sure how I'd go about migrating to a linux server. I have roughly 500 endpoints so I'd rather not start from scratch. Is there a guide somewhere that could help me?

**Reply by si458** [reply]
personally option again: **NEVER ENABLE AUTO-UPDATE!!!**
i use a service like newreleases.io to email me when a new release happens, then i can schedule it upgrade and watch it happen

but anyways in theory so long as you change your DNS entrys/proxy ips etc to the new linux server, you should be fine!
just install meshcentral on linux as normal, then copy the whole meshcentral-data folder over to your linux server!

EDIT: i am so so sorry again for this happening! i just dont no WHY it decided to uninstall the node-windows!?
in theory you can recover, by simply stopping the service, the cd to the meshcentral folder `C:\program files\open source\meshcentral` then just `npm install node-windows` then start the service and it should recover!
but when i was testing after reports, it was a 50/50 that it worked/fixed it!
its all so annoying!!

**Reply by isd94techdir** [reply]
That sounds really easy! I'll give it a shot. Thank you for your work on such an amazing project!

**Reply by si458** [reply]
no worries im just a contributer and im trying to keep it alive!
p.s. dont forget to donate to me to help keep the project alive! https://www.si458.co.uk/2024/01/05/donation/
